core/dataset.cpp

```diff
@@ -1155,77 +1144,11 @@ Dataset histogram(const Dataset &dataset, const Dim &dim) {
   return histogram(dataset, bins);
 }
 
-template <class A, class B>
-void merge_validate_proxies(const A &lhs, const B &rhs) {
-  for (const auto & [ k, v ] : lhs) {
-    /* Throw if an item is present in both proxies but they are not equal */
-    if (rhs.contains(k) && rhs[k] != v) {
-      throw std::runtime_error("Both datasets contain an item with the same "
-                               "key that does not match.");
-    }
-  }
-}
-
-template <class T> void merge_copy_data(Dataset &dest, const T &src) {
-  for (const auto & [ name, data ] : src) {
-    if (!dest.contains(name)) {
-      dest.setData(std::string(name), data);
-    }
-  }
-}
-
-template <class T> void merge_copy_coords(Dataset &dest, const T &src) {
-  for (const auto & [ dim, coord ] : src.coords()) {
-    if (!dest.coords().contains(dim)) {
-      dest.setCoord(dim, coord);
-    }
-  }
-}
-
-template <class T> void merge_copy_labels(Dataset &dest, const T &src) {
-  for (const auto & [ name, label ] : src.labels()) {
-    if (!dest.labels().contains(name)) {
-      dest.setLabels(std::string(name), label);
-    }
-  }
-}
-
-template <class T> void merge_copy_attrs(Dataset &dest, const T &src) {
-  for (const auto & [ name, attr ] : src.attrs()) {
-    if (!dest.attrs().contains(name)) {
-      dest.setAttr(std::string(name), attr);
-    }
-  }
+template <class A, class B> Dataset merge_datasets(const A &a, const B &b) {
```

Comment:
If we now have only a single `merge` function below, can't we call this one `merge` directly without having to go through a `merge_datasets` function?